toast-tk-web-agent/src/main/java/io/toast/tk/agent/web/rest/AsyncHttpClientProvider.java

```diff
@@ -27,15 +40,27 @@ public DefaultAsyncHttpClient get() {
             String proxyPort = this.config.getProxyPort();
             int port = Strings.isNullOrEmpty(proxyPort ) ? -1 : Integer.parseInt(proxyPort);
             ProxyServer proxyServer = new ProxyServer.Builder(config.getProxyAdress(), port).build();
+
+            Builder builder = new DefaultAsyncHttpClientConfig.Builder()
+                    .setProxyServer(proxyServer);
+            
             if(!Strings.isNullOrEmpty(config.getProxyUserName()) && !Strings.isNullOrEmpty(config.getProxyUserPswd()) ){
-                Realm realm = new Realm.Builder(config.getProxyUserName(), config.getProxyUserPswd()).build();
-                AsyncHttpClientConfig cf = new DefaultAsyncHttpClientConfig.Builder()
-                        .setProxyServer(proxyServer).setRealm(realm).build();
-                return new DefaultAsyncHttpClient(cf);
+                Realm realm = new Realm.Builder(config.getProxyUserName(), config.getProxyUserPswd())
+                		.setScheme(AuthScheme.BASIC)
+                		.build();
+                builder.setRealm(realm);
             }
-            AsyncHttpClientConfig cf = new DefaultAsyncHttpClientConfig.Builder()
-                    .setProxyServer(proxyServer).build();
-            return new DefaultAsyncHttpClient(cf);
+            
+			try {
+				if(proxyServer.getHost().startsWith("https")){
+					builder.setSslContext(SslContextBuilder.forClient().trustManager(InsecureTrustManagerFactory.INSTANCE).build());
+				}
+			} catch (SSLException e) {
+				// TODO Auto-generated catch block
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1135)